packages/landing/src/ol.attribution.ts

```diff
@@ -0,0 +1,127 @@
+import { AttributionCollection, AttributionStac } from '@basemaps/shared/build/attribution';
+import { StacLicense } from '@basemaps/shared/build/stac';
+import { View } from 'ol';
+import { Extent } from 'ol/extent';
+import OlMap from 'ol/Map';
+import MapEventType from 'ol/MapEventType';
+import Source from 'ol/source/Source';
+import { Attribution } from './attribution';
+import { MapOptions, MapOptionType, WindowUrl } from './url';
+
+const Copyright = `© ${StacLicense} LINZ`;
+
+async function fetchAttributions(url: string): Promise<AttributionStac | null> {
+    try {
+        const resp = await fetch(url);
+        if (resp.status < 300) {
+            return await resp.json();
+        } else {
+            throw new Error(`fetch attribution failed [${resp.status}] ${resp.statusText}`);
+        }
+    } catch (err) {
+        console.error(err);
+        return null;
+    }
+}
+
+function sameExtent(a: Extent, b: Extent): boolean {
+    if (a.length != b.length) return false;
+    for (let i = 0; i < a.length; ++i) {
+        if (a[i] !== b[i]) return false;
+    }
+    return true;
+}
+
+/**
+ * Handles displaying attributions for the OpenLayers interface
+ */
+export class OlAttribution {
+    source: Source;
+    view: View;
+    config: MapOptions;
+    private _scheduled: any;
+    private _raf = 0;
+    attributions: Attribution | null | undefined;
+    attributionHTML = '';
+    extent: Extent = [0, 0, 0, 0];
+    zoom = -1;
+    filteredRecords: AttributionCollection[] = [];
+
+    /**
+     * Initialize monitoring the OpenLayers map and set the source attributions when changed.
+     */
+    static init(source: Source, map: OlMap, config: MapOptions): void {
+        const attribution = new OlAttribution(source, map.getView(), config);
+
+        map.addEventListener(MapEventType.MOVEEND, () => {
+            attribution.updateAttribution().catch((err) => {
+                console.error(err);
```

Comment:
Do we want to be just logging these to console, or just throw the error?

Reply:
Also if this happens multiple times, does the catch override the previous catchcall?

Reply:
I'll leave the promise rejection unhandled.